### PR TITLE
chore: enable strict apply, bind, and call

### PR DESCRIPTION
**Pull request by mmalerba**
cc @mprobst

### CODE REVIEW

File: src/cdk/drag-drop/directives/drag.spec.ts [hunk 1]
```diff
@@ -1842,14 +1842,22 @@ describe('CdkDrag', () => {
           body: document.body,
           fullscreenElement: document.createElement('div'),
           ELEMENT_NODE: Node.ELEMENT_NODE,
-          querySelectorAll: function() {
-            return document.querySelectorAll.apply(document, arguments);
+          querySelectorAll: function(...args: [string]) {
```
Comment (crisbeto) [inline src/cdk/drag-drop/directives/drag.spec.ts]: Shouldn't it be `string[]`?

File: src/cdk/overlay/fullscreen-overlay-container.spec.ts [hunk 2]
```diff
@@ -27,28 +27,28 @@ describe('FullscreenOverlayContainer', () => {
...
               if (eventName === 'fullscreenchange') {
                 fullscreenListeners.delete(listener);
               } else {
-                document.addEventListener.apply(document, arguments);
+                document.addEventListener(eventName, listener);
               }
             },
-            querySelectorAll: function() {
-              return document.querySelectorAll.apply(document, arguments);
+            querySelectorAll: function(...args: [string]) {
```
Comment (crisbeto) [inline src/cdk/overlay/fullscreen-overlay-container.spec.ts]: Also should be `string[]`? There are a few more changes like this.

File: src/cdk/drag-drop/directives/drag.spec.ts (hunk 1 above)
Comment (mmalerba) [reply src/cdk/drag-drop/directives/drag.spec.ts]: `string[]` is a string array of any length, `[string]` is an array of exactly one string

Comment (crisbeto) [reply src/cdk/drag-drop/directives/drag.spec.ts]: If we know it'll be one string, couldn't we just declare it like this?

```
querySelectorAll: (selector: string) {
  return document.querySelectorAll.call(document, selector);
}
```

Comment (mmalerba) [reply src/cdk/drag-drop/directives/drag.spec.ts]: My thinking was that for these ones where we're just wrapping a function and passing things through, the author was trying to be defensive against future browser API changes by just grabbing all of the arguments and shoving them through. This style preserves that

Comment (crisbeto) [reply src/cdk/drag-drop/directives/drag.spec.ts]: I'm pretty sure that I'm the author and I think I did it because it was shorter 😄

Comment (devversion) [reply src/cdk/drag-drop/directives/drag.spec.ts]: I can only speak for the ZoneJS thing where I did it that way. I wanted to be on the safe side if a different ZoneJS version is used / when the signature changes. The typings are manually copied.

Comment (mmalerba) [reply src/cdk/drag-drop/directives/drag.spec.ts]: Yeah `querySelectorAll` probably isn't going to change much, but some do, `addEventListener` has definitely changed over the years

**Pull request merged**